src/Mvc/Mvc.Core/src/Formatters/InputFormatter.cs

```diff
@@ -98,8 +99,11 @@ public virtual Task<InputFormatterResult> ReadAsync(InputFormatterContext contex
             throw new ArgumentNullException(nameof(context));
         }
 
-        var request = context.HttpContext.Request;
-        if (request.ContentLength == 0)
+        var hasBody = context.HttpContext.Features.Get<IHttpRequestBodyDetectionFeature>()?.CanHaveBody;
+        // In case the feature is not registered
+        hasBody ??= context.HttpContext.Request.ContentLength != 0;
+
+        if (hasBody == false)
```

Comment:
nit:
```suggestion
        if (!hasBody)
```

Reply:
It's a `Nullable<bool>`. This is basically a saying "if we know for sure this request cannot have a body ..." Probably worth a comment.